tests/test_georaster.py

```diff
@@ -379,11 +379,27 @@ def test_crop(self) -> None:
 
     def test_reproj(self) -> None:
 
-        # Test reprojecting to dst_ref
+        # Reference raster to be used
         r = gr.Raster(datasets.get_path("landsat_B4"))
         r.set_ndv(0)  # to avoid warnings - will be used when reprojecting outside bounds
+
+        # A second raster with different bounds, shape and resolution
         r2 = gr.Raster(datasets.get_path("landsat_B4_crop"))
+        r2 = r2.reproject(dst_res=20)
+        assert r2.res == (20, 20)
```

Comment:
Mypy complains here because you compare floats with integers, the best fix is:
```python
assert r2.res == (20., 20.)
```
or if you (understandably) don't like this behaviour:
```python
assert r2.res == (20, 20)  # type: ignore
```

Reply:
Thanks for the feedback! I guess another solution is to indicate that res could also be a tuple of int. Would that be
`list[float | int, float | int]` ?

Reply:
The version with `Number` does not seem to work, but `tuple[float | int, float | int]` yes.

Reply:
`tuple[float | int, float | int]` would accept inconsistent types:
`(5, 5.5)`
`(3.3, 3)`
etc.

If this is fine by us, the type annotation is okay! Otherwise, it would be:
```python
tuple[float, float] | tuple[int, int]
```
I'm fine with either